liteipc.lua

```diff
@@ -1,115 +1,39 @@
 -- liteipc - async IPC for lite
 
-local fs = require "plugins.lintplus.fsutil"
+local liteipc = {}
 
-local liteipc_loader = {}
+local Process = {}
+Process.__index = Process
 
--- a fallback to support io.popen in case the user didn't install the
--- async process runtime
-function liteipc_loader.sync()
-
-  local liteipc = {}
-
-  local Process = {}
-  Process.__index = Process
-
-  -- based on nim stdlib: os.quoteShellWindows, os.quoteShellPosix
-
-  local quote_shell
-  if PLATFORM == "Windows" then
-    quote_shell = function (arg)
-      local need_quote = #arg == 0 or arg:match("[ \t]") ~= nil
-      local result = ""
-      local backslash_buffer = {}
-      if need_quote then
-        result = result .. '"'
-      end
-      for i = 1, #arg do
-        local c = arg:sub(i, i)
-        if c == '\\' then
-          table.insert(backslash_buffer, c)
-        elseif c == '"' then
-          local buffer_concat = table.concat(backslash_buffer)
-          result = result ..
-            buffer_concat .. buffer_concat .. -- wtf?
-            '\\"'
-          backslash_buffer = {}
-        else
-          if #backslash_buffer > 0 then
-            result = result .. table.concat(backslash_buffer)
-            backslash_buffer = {}
-          end
-          result = result .. c
-        end
-      end
-      if need_quote then
-        result = result .. '"'
-      end
-      return result
-    end
-  else
-    quote_shell = function (arg)
-      if #arg == 0 then
-        return "''"
-      end
-      local is_safe = arg:match("[^%%%+%-%./_:=@0-9a-zA-Z]") == nil
-      if is_safe then
-        return arg
-      else
-        return "'" .. arg:gsub("'", "'\"'\"'") .. "'"
-      end
-    end
-  end
-
-  local function escape_args(args)
-    local result = {}
-    for _, arg in ipairs(args) do
-      table.insert(result, quote_shell(arg))
-    end
-    return table.concat(result, ' ')
-  end
+function liteipc.start_process(args, cwd)
+  local proc = setmetatable({
+    popen = process.start(args, cwd and {cwd = cwd} or {})
```

Comment:
If `cwd` is nil, then `{cwd = cwd}` will construct an empty table (because `nil` in a table means that there's no element at that position).

Reply:
you are right, changed to just `{cwd = cwd}`